### DISCUSSION TITLE
Respawn multiple objects simultaneously

### DISCUSSION THREAD

**Discussion by Snowblaze** [enable3d/enable3d#261] [Help / Questions] [answered]
Hey. I'm trying to write a 1 on 1 shooter with the headless package and I found myself in quite a pickle. When one of the players' hp gets to 0, I try to respawn them with the following code.

```
for (let i = 0; i < this.players.length; i += 1) {
  const player = this.players[i];
  player.hitPoints = 100;
  player.gameObject.body.transform();
  player.gameObject.body.setVelocity(0, 0, 0);
  player.gameObject.body.setAngularVelocity(0, 0, 0);
  player.gameObject.body.setCollisionFlags(2);
  player.gameObject.body.setPosition(
    spawnPoints[i].x,
    spawnPoints[i].y,
    spawnPoints[i].z,
  );
  player.gameObject.body.refresh();
  player.gameObject.body.once.update(() => {
    player.gameObject.body.setCollisionFlags(0);
  });
}
```

The problem is that update just doesn't work, I tried logging anything in the update and still got nothing. Am I missing something? Would appreciate a bit of help!

**Comment by yandeu** [comment, marked as answer]
`body.once.update()` does never get called?

What if you add `.body.needUpdate = true` just after `.body.refresh()`?

---

See the teleport example: https://enable3d.io/examples/teleport-a-dynamic-body.html

**Reply by Snowblaze** [reply]
After doing that, the update worked, thanks a lot! But now it seems that I need to change the position on the extended object instead of the body, right? I tried that and it worked like a charm. So can you clarify when setPosition works on the body and when one should change the position of the extended object?

**Reply by yandeu** [reply]
Once you set collisionFlag to kinematic, it's on the object. Once you change it back to dynamic, it's on the body again.

See: https://enable3d.io/docs.html#physics-body
